src/components/ArticleList.jsx

```diff
@@ -0,0 +1,57 @@
+import { getAllArticles } from "../../api";
+import React, { useState, useEffect } from "react";
+
+const ArticleList = () => {
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setError(null);
+    getAllArticles()
+      .then((data) => {
+        setArticles(data.articles);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setError("error fetching data");
+        setIsLoading(false);
+      });
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading items....</p>;
+  }
+  if (error) {
+    return <p>Something went wrong...</p>;
```

Comment:
Since you've created an error state it would be nice to display the error message here instead of a generic something went wrong.